modin/core/dataframe/pandas/utils.py

```diff
@@ -44,3 +47,34 @@ def concatenate(dfs):
                 i, pandas.Categorical(df.iloc[:, i], categories=union.categories)
             )
     return pandas.concat(dfs)
+
+
+def merge_partitioning(left, right, axis=1):
+    """
+    Get the number of splits across the `axis` for the two dataframes being concatenated.
+
+    Parameters
+    ----------
+    left : PandasDataframe
+    right : PandasDataframe
+    axis : int, default: 1
+
+    Returns
+    -------
+    int
+    """
+    # Avoiding circular imports from pandas query compiler
+    from modin.core.storage_formats.pandas.utils import compute_chunksize
+
+    lsplits = left._partitions.shape[axis]
+    rsplits = right._partitions.shape[axis]
+
+    lshape = left._row_lengths_cache if axis == 0 else left._column_widths_cache
+    rshape = right._row_lengths_cache if axis == 0 else right._column_widths_cache
```

Comment:
```suggestion
    lshape = left._partitions.shape[axis]
    rshape = right._partitions.shape[axis]

    lsplits = left._row_lengths_cache if axis == 0 else left._column_widths_cache
    rsplits = right._row_lengths_cache if axis == 0 else right._column_widths_cache
```
Does this make sense?

Also, we should probably move the upper variables to the else branch because they are used only there.

Reply:
don't think the renaming is needed, `_partitions.shape[axis]` clearly indicates the number of splits, and the row/column widths caches are for partition's shapes along the axis. For me, it seems that the naming is accurate as it is.

> Also, we should probably move the upper variables to the else branch because they are used only there.

done